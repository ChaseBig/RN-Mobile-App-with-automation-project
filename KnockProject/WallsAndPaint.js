import React, { Component } from 'react';
import { StyleSheet, TouchableOpacity, Text, View } from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';
import { CameraLauncher } from './CameraLauncher';
import { CommentsLauncher } from './CommentsLauncher';

const wallCondition = {
    Fix: 'Fix',
    Poor: 'Poor',
    Average: 'Average',
    Good: 'Good',
    Excellent: 'Excellent'
}

export class WallsAndPaint extends Component {
    constructor(props){
        super(props)
        this.onCameraLaunchButtonPress = this.onCameraLaunchButtonPress.bind(this);
        this.onCommentsButtonPress = this.onCommentsButtonPress.bind(this);
        this.state = {
            wallCondition,
            isCameraVisible: false,
            isShowingCommentsModal: false
        }
    }
    render() {
        return (
            <View style={ styles.container }>
                <Text style={{fontWeight: 'bold', fontSize: 15}}>Walls / Paint</Text>
                    <View style={styles.buttonGroup} >
                        <TouchableOpacity testID='floorLevelSelection' onPress={() => this.setState({ wallCondition: wallCondition.Fix })}>
                            <Icon
                                name='medkit'
                                size={35}
                                style={this.state.wallCondition === (wallCondition.Fix) ? styles.btnActive : styles.btnInactive}
                            />
                            <Text style={{marginLeft: 21, fontSize:11}}>{wallCondition.Fix}</Text>
                        </TouchableOpacity>
                        <TouchableOpacity testID='floorLevelSelection' onPress={() => this.setState({wallCondition: wallCondition.Poor})}>
                            <Icon
                                name='frown-o'
                                size={35}
                                style={this.state.wallCondition === (wallCondition.Poor) ? styles.btnActive : styles.btnInactive}
                            />
                            <Text style={{marginLeft: 22, paddingRight: -10, fontSize:11}}>{wallCondition.Poor}</Text>
                         </TouchableOpacity>
                        <TouchableOpacity testID='floorLevelSelection' onPress={() => this.setState({wallCondition: wallCondition.Average})}>
                            <Icon
                                name='meh-o'
                                size={35}
                                style={this.state.wallCondition === (wallCondition.Average) ? styles.btnActive : styles.btnInactive}
                            />
                            <Text style={{marginLeft: 15, paddingLeft: 10, fontSize:11}}>{wallCondition.Average}</Text>
                        </TouchableOpacity>
                        <TouchableOpacity testID='floorLevelSelection' onPress={() => this.setState({ wallCondition: wallCondition.Good })}>
                            <Icon
                                name='smile-o'
                                size={35}
                                style={this.state.wallCondition === (wallCondition.Good) ? styles.btnActive : styles.btnInactive}
                            />
                            <Text style={{marginLeft: 11, paddingLeft: 1,fontSize:11}}>{wallCondition.Good}</Text>
                        </TouchableOpacity>
                        <TouchableOpacity testID='floorLevelSelection' onPress={() => this.setState({ wallCondition: wallCondition.Excellent })}>
                            <Icon
                                name='smile-o'
                                size={35}
                                style={this.state.wallCondition === (wallCondition.Excellent) ? styles.btnActive : styles.btnInactive}
                            />
                        <Text style={{marginLeft: 11,fontSize:11}}>{wallCondition.Excellent}</Text>
                        </TouchableOpacity>
                    <View>
                        <CameraLauncher onPress={this.onCameraLaunchButtonPress} />
                        <CommentsLauncher onPress={this.onCommentsButtonPress} />
                    </View>
                </View>
            </View>
        );
    }

onCameraLaunchButtonPress(isCameraVisible){
    this.setState({
        isCameraVisible: true
    });
  }


onCommentsButtonPress(isShowingCommentsModal){
    this.setState({
        isShowingCommentsModal: true
    });
  }
}

const styles = StyleSheet.create({
    container: {
        marginTop: 35, 
        textAlign: 'right',
        marginRight: -100
    },
    buttonGroup: {
        lineHeight: 13,
        flexDirection: 'row',
        marginTop: 32,
        marginLeft: 1
    },
    text: {
        textAlign: 'center',
        fontSize: 25,
        marginBottom: 30,
        justifyContent: 'center',
        alignItems: 'center',
        alignSelf: 'center',
        flexDirection: 'row'
    },
    btnActive: {
        color: 'green',
        marginLeft: 19,
        marginRight: 3,
        paddingLeft: 0,
        paddingRight: -6
    },
    btnInactive: {
        color: '#B5B5B5',
        marginLeft: 19,
        marginRight: 3,
        paddingLeft: 0,
        paddingRight: -6
    }
});