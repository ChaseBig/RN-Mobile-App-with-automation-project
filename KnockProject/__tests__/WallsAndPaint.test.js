import React from 'react';
import { shallow, mount } from 'enzyme';
import { WallsAndPaint } from '../WallsAndPaint';
import { TouchableOpacity } from 'react-native';
import renderer from 'react-native-test-utils';
import mockComponent from 'react-native/jest/mockComponent';
import styled from 'styled-components/native';

const StyledComponent = styled.TouchableOpacity``;

describe('WallsAndPaint', () => {
  it('should render correctly with no props', () => {
    const component = shallow(<WallsAndPaint/>);
    
    expect(component).toMatchSnapshot();
  });

  it('should render view container text correctly with given string', () => {
    const strings = ['Walls / Paint'];
    const component = shallow(<WallsAndPaint list={strings} />);
    expect(component).toMatchSnapshot();
  });

  it('should render TouchableOpacity "Good" wall condition', () => {
    jest.mock('TouchableOpacity', () => mockComponent('TouchableOpacity'));
    const mockFn = jest.fn();
    const Component = <TouchableOpacity onPress={mockFn}>Fix</TouchableOpacity>;
    const tree = renderer(Component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should invoke press event on TouchableOpacity "Good" wall condition', () => {
    jest.mock('TouchableOpacity', () => mockComponent('TouchableOpacity'));
    const mockFn = jest.fn();
    const Component = <TouchableOpacity onPress={mockFn}>Fix</TouchableOpacity>;
    const dom = renderer(Component);

    expect(mockFn).not.toHaveBeenCalled();
    dom.simulate('press');
    expect(mockFn).toHaveBeenCalled();
  });
});