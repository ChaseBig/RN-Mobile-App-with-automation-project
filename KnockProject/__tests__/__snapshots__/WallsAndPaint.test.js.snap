// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WallsAndPaint should render TouchableOpacity "Good" wall condition 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[MockFunction]}
>
  Fix
</TouchableOpacity>
`;

exports[`WallsAndPaint should render correctly with no props 1`] = `
<Component
  style={
    Object {
      "marginRight": -100,
      "marginTop": 35,
      "textAlign": "right",
    }
  }
>
  <Component
    style={
      Object {
        "fontSize": 15,
        "fontWeight": "bold",
      }
    }
  >
    Walls / Paint
  </Component>
  <Component
    style={
      Object {
        "flexDirection": "row",
        "lineHeight": 13,
        "marginLeft": 1,
        "marginTop": 32,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      testID="floorLevelSelection"
    >
      <Icon
        allowFontScaling={false}
        name="medkit"
        size={35}
        style={
          Object {
            "color": "#B5B5B5",
            "marginLeft": 19,
            "marginRight": 3,
            "paddingLeft": 0,
            "paddingRight": -6,
          }
        }
      />
      <Component
        style={
          Object {
            "fontSize": 11,
            "marginLeft": 21,
          }
        }
      >
        Fix
      </Component>
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      testID="floorLevelSelection"
    >
      <Icon
        allowFontScaling={false}
        name="frown-o"
        size={35}
        style={
          Object {
            "color": "#B5B5B5",
            "marginLeft": 19,
            "marginRight": 3,
            "paddingLeft": 0,
            "paddingRight": -6,
          }
        }
      />
      <Component
        style={
          Object {
            "fontSize": 11,
            "marginLeft": 22,
            "paddingRight": -10,
          }
        }
      >
        Poor
      </Component>
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      testID="floorLevelSelection"
    >
      <Icon
        allowFontScaling={false}
        name="meh-o"
        size={35}
        style={
          Object {
            "color": "#B5B5B5",
            "marginLeft": 19,
            "marginRight": 3,
            "paddingLeft": 0,
            "paddingRight": -6,
          }
        }
      />
      <Component
        style={
          Object {
            "fontSize": 11,
            "marginLeft": 15,
            "paddingLeft": 10,
          }
        }
      >
        Average
      </Component>
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      testID="floorLevelSelection"
    >
      <Icon
        allowFontScaling={false}
        name="smile-o"
        size={35}
        style={
          Object {
            "color": "#B5B5B5",
            "marginLeft": 19,
            "marginRight": 3,
            "paddingLeft": 0,
            "paddingRight": -6,
          }
        }
      />
      <Component
        style={
          Object {
            "fontSize": 11,
            "marginLeft": 11,
            "paddingLeft": 1,
          }
        }
      >
        Good
      </Component>
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      testID="floorLevelSelection"
    >
      <Icon
        allowFontScaling={false}
        name="smile-o"
        size={35}
        style={
          Object {
            "color": "#B5B5B5",
            "marginLeft": 19,
            "marginRight": 3,
            "paddingLeft": 0,
            "paddingRight": -6,
          }
        }
      />
      <Component
        style={
          Object {
            "fontSize": 11,
            "marginLeft": 11,
          }
        }
      >
        Excellent
      </Component>
    </TouchableOpacity>
    <Component>
      <CameraLauncher
        onPress={[Function]}
      />
      <CommentsLauncher
        onPress={[Function]}
      />
    </Component>
  </Component>
</Component>
`;

exports[`WallsAndPaint should render view container text correctly with given string 1`] = `
<Component
  style={
    Object {
      "marginRight": -100,
      "marginTop": 35,
      "textAlign": "right",
    }
  }
>
  <Component
    style={
      Object {
        "fontSize": 15,
        "fontWeight": "bold",
      }
    }
  >
    Walls / Paint
  </Component>
  <Component
    style={
      Object {
        "flexDirection": "row",
        "lineHeight": 13,
        "marginLeft": 1,
        "marginTop": 32,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      testID="floorLevelSelection"
    >
      <Icon
        allowFontScaling={false}
        name="medkit"
        size={35}
        style={
          Object {
            "color": "#B5B5B5",
            "marginLeft": 19,
            "marginRight": 3,
            "paddingLeft": 0,
            "paddingRight": -6,
          }
        }
      />
      <Component
        style={
          Object {
            "fontSize": 11,
            "marginLeft": 21,
          }
        }
      >
        Fix
      </Component>
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      testID="floorLevelSelection"
    >
      <Icon
        allowFontScaling={false}
        name="frown-o"
        size={35}
        style={
          Object {
            "color": "#B5B5B5",
            "marginLeft": 19,
            "marginRight": 3,
            "paddingLeft": 0,
            "paddingRight": -6,
          }
        }
      />
      <Component
        style={
          Object {
            "fontSize": 11,
            "marginLeft": 22,
            "paddingRight": -10,
          }
        }
      >
        Poor
      </Component>
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      testID="floorLevelSelection"
    >
      <Icon
        allowFontScaling={false}
        name="meh-o"
        size={35}
        style={
          Object {
            "color": "#B5B5B5",
            "marginLeft": 19,
            "marginRight": 3,
            "paddingLeft": 0,
            "paddingRight": -6,
          }
        }
      />
      <Component
        style={
          Object {
            "fontSize": 11,
            "marginLeft": 15,
            "paddingLeft": 10,
          }
        }
      >
        Average
      </Component>
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      testID="floorLevelSelection"
    >
      <Icon
        allowFontScaling={false}
        name="smile-o"
        size={35}
        style={
          Object {
            "color": "#B5B5B5",
            "marginLeft": 19,
            "marginRight": 3,
            "paddingLeft": 0,
            "paddingRight": -6,
          }
        }
      />
      <Component
        style={
          Object {
            "fontSize": 11,
            "marginLeft": 11,
            "paddingLeft": 1,
          }
        }
      >
        Good
      </Component>
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      testID="floorLevelSelection"
    >
      <Icon
        allowFontScaling={false}
        name="smile-o"
        size={35}
        style={
          Object {
            "color": "#B5B5B5",
            "marginLeft": 19,
            "marginRight": 3,
            "paddingLeft": 0,
            "paddingRight": -6,
          }
        }
      />
      <Component
        style={
          Object {
            "fontSize": 11,
            "marginLeft": 11,
          }
        }
      >
        Excellent
      </Component>
    </TouchableOpacity>
    <Component>
      <CameraLauncher
        onPress={[Function]}
      />
      <CommentsLauncher
        onPress={[Function]}
      />
    </Component>
  </Component>
</Component>
`;
